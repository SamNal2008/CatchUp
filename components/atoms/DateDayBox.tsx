import {useColorSchemeOrDefault} from "@/hooks/useColorScheme";
import {View} from "react-native";
import {Colors} from "@/constants/design";
import {ThemedText} from "@/components/atoms/ThemedText";
import React from "react";


const getWeekDayName = (day: number) => {
    switch (day) {
        case 0:
            return 'Sunday';
        case 1:
            return 'Monday';
        case 2:
            return 'Tuesday';
        case 3:
            return 'Wednesday';
        case 4:
            return 'Thursday';
        case 5:
            return 'Friday';
        case 6:
            return 'Saturday';
    }
}

export const DateDayBox = ({date}: { date: Date }) => {
    const weekDay = getWeekDayName(date.getDay())?.toUpperCase().substring(0, 3);
    const theme = useColorSchemeOrDefault();
    return (
        <View style={{
            height: 40,
            backgroundColor: Colors[theme].background,
            alignItems: 'center',
            borderRadius: 6,
            paddingHorizontal: 8,
        }}>
            <ThemedText style={{
                height: 18,
                fontFamily: 'SF Pro',
                fontSize: 13,
                lineHeight: 18,
            }}>
                {weekDay}
            </ThemedText>
            <ThemedText style={{
                width: 26,
                height: 22,
                fontFamily: 'SF Pro',
                fontWeight: '600',
                fontSize: 17,
                lineHeight: 22,
                textAlign: 'center'
            }}>
                {date.getDate()}
            </ThemedText>
        </View>
    );
}
