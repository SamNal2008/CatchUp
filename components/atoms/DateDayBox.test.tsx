import { render, screen } from "@testing-library/react-native";
import { DateDayBox } from "./DateDayBox";

describe("DateDayBox", () => {
  const mockDate = new Date("2025-01-05");

  it("matches snapshot", () => {
    const { toJSON } = render(<DateDayBox date={mockDate} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("matches snapshot with custom style", () => {
    const { toJSON } = render(<DateDayBox date={mockDate} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("renders the correct day and month", () => {
    render(<DateDayBox date={mockDate} />);

    // Using getByText with a regular expression to match the day (5)
    expect(screen.getByText("5")).toBeTruthy();

    // Check for the month (Jan)
    expect(screen.getByText("SUN")).toBeTruthy();
  });
});
