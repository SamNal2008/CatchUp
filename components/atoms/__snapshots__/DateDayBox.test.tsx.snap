// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateDayBox matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#F3F2F8",
      "borderRadius": 6,
      "height": 40,
      "paddingHorizontal": 8,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#000000",
        },
        {
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        {
          "fontFamily": "SF Pro",
          "fontSize": 13,
          "height": 18,
          "lineHeight": 18,
        },
      ]
    }
  >
    SUN
  </Text>
  <Text
    style={
      [
        {
          "color": "#000000",
        },
        {
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        {
          "fontFamily": "SF Pro",
          "fontSize": 17,
          "fontWeight": "600",
          "height": 22,
          "lineHeight": 22,
          "textAlign": "center",
          "width": 26,
        },
      ]
    }
  >
    5
  </Text>
</View>
`;

exports[`DateDayBox matches snapshot with custom style 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#F3F2F8",
      "borderRadius": 6,
      "height": 40,
      "paddingHorizontal": 8,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#000000",
        },
        {
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        {
          "fontFamily": "SF Pro",
          "fontSize": 13,
          "height": 18,
          "lineHeight": 18,
        },
      ]
    }
  >
    SUN
  </Text>
  <Text
    style={
      [
        {
          "color": "#000000",
        },
        {
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        {
          "fontFamily": "SF Pro",
          "fontSize": 17,
          "fontWeight": "600",
          "height": 22,
          "lineHeight": 22,
          "textAlign": "center",
          "width": 26,
        },
      ]
    }
  >
    5
  </Text>
</View>
`;
